packages/integration-karma/test/shadow-dom/multiple-templates/index.spec.js

```diff
@@ -69,24 +69,13 @@ describe('multiple stylesheets rendered in same component', () => {
                 return new Promise((resolve) => requestAnimationFrame(() => resolve()));
             })
             .then(() => {
-                if (process.env.NATIVE_SHADOW) {
-                    // TODO [#2466]: In native shadow, stylesheets are not removed from the DOM
-                    expect(
-                        getComputedStyle(element.shadowRoot.querySelector('.red')).color
-                    ).toEqual('rgb(255, 0, 0)');
-                    expect(
-                        getComputedStyle(element.shadowRoot.querySelector('.green')).color
-                    ).toEqual('rgb(0, 128, 0)');
-                    expect(getComputedStyle(element).marginLeft).toEqual('5px');
-                } else {
-                    expect(
-                        getComputedStyle(element.shadowRoot.querySelector('.red')).color
-                    ).toEqual('rgb(0, 0, 0)');
-                    expect(
-                        getComputedStyle(element.shadowRoot.querySelector('.green')).color
-                    ).toEqual('rgb(0, 0, 0)');
-                    expect(getComputedStyle(element).marginLeft).toEqual('0px');
-                }
+                expect(getComputedStyle(element.shadowRoot.querySelector('.red')).color).toEqual(
+                    'rgb(0, 0, 0)'
+                );
+                expect(getComputedStyle(element.shadowRoot.querySelector('.green')).color).toEqual(
+                    'rgb(0, 0, 0)'
+                );
+                expect(getComputedStyle(element).marginLeft).toEqual('0px');
```

Comment:
This is a good example of the behavior becoming more consistent with this PR. This is very subtle, because what's actually happening here is that, in synthetic, the stylesheets are also not removed from the DOM. But the reason the color is "correct" in synthetic mode is that the scope attributes are not applied to the DOM nodes (due to the `stylesheets` array being empty).

In any case, this PR makes the behavior consistent.